pkg/controllers/plugin/plugin_workload_status_controller.go

```diff
@@ -0,0 +1,240 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	greenhousev1alpha1 "github.com/cloudoperators/greenhouse/pkg/apis/greenhouse/v1alpha1"
+	"github.com/cloudoperators/greenhouse/pkg/clientutil"
+	"github.com/cloudoperators/greenhouse/pkg/helm"
+)
+
+const (
+	StatusRequeueInterval          = 2 * time.Minute
+	helmReleaseNameAnnotation      = "meta.helm.sh/release-name"
+	helmReleaseNamespaceAnnotation = "meta.helm.sh/release-namespace"
+	jobNameLabel                   = "batch.kubernetes.io/job-name"
+	monitoringAPIGroupName         = "monitoring.coreos.com"
+)
+
+type ReleaseStatus struct {
+	ReleaseName         string `json:"releaseName,omitempty" protobuf:"bytes,1,opt,name=releaseName"`
+	ReleaseNamespace    string `json:"releaseNamespace,omitempty" protobuf:"bytes,3,opt,name=namespace"`
+	HelmStatus          string `json:"helmStatus,omitempty"`
```

Comment:
More fine-grained information would be exposed via `prometheus metrics`. like 4 scheduled 4 available ...
Even more fine-grained like status like: individual status of pods belongs to a deployment was deprecated as agreed with @IvoGoman since we agreed to keep information from the "high level" objects only